x/mongo/driver/topology/connection.go

```diff
@@ -142,6 +142,8 @@ func (c *connection) connect(ctx context.Context) {
 	}
 	defer close(c.connectDone)
 
+	_, isOperationDeadline := ctx.Deadline()
```

Comment:
This will still be true if ctx.Deadline() is later than connectTimeout, which I don't think we want

Reply:
Per discussion in the sync meeting, we no longer use the existence of a deadline to indicate an operation-scoped timeout happened. Instead, we check for any errors on the passed in `Context`.